atlite/resource.py

```diff
@@ -29,17 +29,45 @@
 import numpy as np
 from scipy.signal import fftconvolve
 from pkg_resources import resource_stream
+import ast
+
+from . import config
+from .wind import download_turbineconf
 
 import logging
 logger = logging.getLogger(name=__name__)
 
 def get_windturbineconfig(turbine):
-    """Load the 'turbine'.yaml file from local disk and provide a turbine dict."""
-
-    res_name = "resources/windturbine/" + turbine + ".yaml"
-    turbineconf = yaml.safe_load(resource_stream(__name__, res_name))
-    V, POW, hub_height = itemgetter('V', 'POW', 'HUB_HEIGHT')(turbineconf)
-    return dict(V=np.array(V), POW=np.array(POW), hub_height=hub_height, P=np.max(POW))
+    """Load the wind 'turbine' configuration.
+
+    The configuration can either be one from local storage, then 'turbine' is
+    considered part of the file base name '<turbine>.yaml' in config.windturbine_dir.
+    Alternatively 'turbine' is the string representation of a dict refer to a unique
+    turbine OEDB.
+
+    Parameter
+    ---------
+    turbine : str
+        Name of the local turbine file (without .yaml suffix) in config.windturbine_dir
+        or a string starting with "oedb:" for an online lookup in the OEDB database.
+        The online database will be searched for a turbine according to the dict following
+        the flag, e.g. "oedb:{'id':'10'}".
+        For all supported fields, see atlite.wind.download_turbineconf() for details.
+    """
+
+    if turbine.startswith('oedb:'):
+        source = 'oedb:'
+        turbine = turbine[5:] # Remove 'oedb:' switch
+        turbine = ast.literal_eval(turbine)
+
+        return download_turbineconf(turbine, store_locally=False)
```

Comment:
Either `download_turbineconf` should raise an error, or we have to raise an error if we encounter `None` here. Otherwise, `cutout.wind` will suddenly return a layout rather than a time-series, which is definitely unwanted!

Reply:
We can raise a `AmbigiouslySpecifiedTurbineException`.

If we continue the process with `None`, then I think the conversion will just fail unexpectedly instead of generating a layout.

I simplified it a bit further, removed 'oedb:' from the string argument in favour of a argument `source='local'|'oedb'`.

Reply:
I still like the `oedb:` prefix together with the possibility to use simple strings for the most usual queries. If you want to keep your `source` argument, you should also provide a way to accept it in `cutout.wind`.